src/ClientGenerator/NamespaceGenerator.cs

```diff
@@ -392,7 +392,7 @@ protected string GetActualMethodReturnType(Type type, SerializeFlag flag)
 
                 var errorMsg = string.Format("Unexpected number of arguments {0} for generic type {1} used as a return type. Only Type<T> are supported as generic return types of grain methods.", genericArguments.Length, type);
                 ReportError(errorMsg);
-                throw new ApplicationException(errorMsg);
+                throw new ArgumentException(errorMsg);
```

Comment:
Another minor thing - per .NET design guidelines it should be `throw new ArgumentException(errorMsg, "type");` to indicate which argument is invalid.